File: typescript/encryption/domain/encryption.ts
import {CharReader} from "./char_reader";
import {CharWriter} from "./char_writer";

export class Encryption {
    reader: CharReader
    writer: CharWriter
    constructor(reader: CharReader, writer: CharWriter) {
        this.reader = reader
        this.writer = writer
    }

    encryption() {
        const input = this.reader.read()
        const encrypted = this.translate(input)
        this.writer.write(encrypted)
    }

    private translate(input: string): string {
        return input
    }
}

File: go/encryption/domain/encryption.go
package domain

type CharReader interface {
	Read() string
}

type CharWriter interface {
	Write(string)
}

func Encryption(reader CharReader, writer CharWriter) {
	input := reader.Read()
	encrypted := translate(input)
	writer.Write(encrypted)
}

func translate(input string) string {
	return input
}

